src/lib.rs

```diff
@@ -1432,10 +1437,10 @@ impl Fee {
     }
 
     /// Compute the fee if possible (it can fail in case the values are out of range)
-    pub fn calculate(&self, tx: Transaction) -> Option<Value> {
+    pub fn calculate(&self, tx: &Transaction) -> Option<Value> {
         use EitherTransaction::Certificate;
         use EitherTransaction::NoCertificate;
-        match (&self.0, tx.0) {
+        match (&self.0, tx.0.clone()) {
```

Comment:
You can just use `match (&self.0, &tx.0)`. There is no need to clone here, just changing the thing to borrow is enough